.editorconfig

```diff
@@ -0,0 +1,20 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+
+[*.{cpp,py,ts}]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[*.md]
+indent_style = space
+indent_size = 4
```

Comment:
There are some inconsistencies in existing markdown files (a few bullet lists use 1 or 2 spaces) but the vast majority uses 4 spaces. We can settle on 2 or 4 spaces, up to you.